### DISCUSSION TITLE
How can I access or send the metadata in a function tool?

### DISCUSSION THREAD

**Discussion by cvillegas02** [RageAgainstThePixel/OpenAI-DotNet#362] [Q&A] [answered]
Thank you very much for your open ia library, the problem I have is that I have not found a way to send or obtain the metadata, such as a user ID that I need in the function that I call in a certain thread.

![image](https://github.com/user-attachments/assets/52d6768e-699d-4597-b1c2-1491c532be80)

![image](https://github.com/user-attachments/assets/6ff4d410-73fb-4281-87d4-bb5b04bba88f)

**Comment by StephenHodgson** [comment, marked as answer]
It depends on how you've implemented your tool and the properties you are exposing.

The metadata is whatever you wish to make for it. Then you fill in that data as you process the tool results and return it to the assistant.

There is also metadata you can add to the assistant, those are in the OpenAI spec, but the rest is up to you and your data models you're already using.

**Reply by cvillegas02** [reply]
Gracias por tu respuesta.

Saludos...
